### PR TITLE
fix: Added error checks when requesting stats fails

### PR DESCRIPTION
**Pull request by carlosroman**

### CODE REVIEW

File: src/metrics_test.go [hunk 1]
```diff
@@ -148,3 +158,70 @@ func Test_pathToPrefix(t *testing.T) {
 		})
 	}
 }
+
+func Test_getMetricsData(t *testing.T) {
```
Comment (ardias) [inline src/metrics_test.go]: I'm really starting to hate this go way of testing 😢

File: src/metrics_test.go [hunk 2]
```diff
@@ -148,3 +158,70 @@ func Test_pathToPrefix(t *testing.T) {
...
+			args.StatusURL = ts.URL
+			err = getMetricsData(ms)
+			t.Log(err)
+			if tt.expectErr != nil {
+				assert.EqualError(t, err, tt.expectErr.Error())
+			} else {
+				assert.NoError(t, err)
+				t.Log(ms)
+				assert.Equal(t, tt.expectedConnectionsActive, ms.Metrics["net.connectionsActive"])
+			}
```
Comment (Noly) [inline src/metrics_test.go]: The 't.Log(x)' are intentional?

File: src/metrics_test.go [hunk 3]
```diff
@@ -148,3 +158,70 @@ func Test_pathToPrefix(t *testing.T) {
 		})
 	}
 }
+
+func Test_getMetricsData(t *testing.T) {
+	tests := []struct {
```
Comment (Noly) [inline src/metrics_test.go]: I prefer to have one test for each case, is more easy to read and express intention.

Comment (Noly) [reply src/metrics_test.go]: At first I thought we were testing happy path more than errors handling

File: src/metrics_test.go (hunk 2 above)
Comment (carlosroman) [reply src/metrics_test.go]: Just wanted to log stuff in the test for when things fail. Makes it easier to debug the test.

**Pull request merged**